src/snowflake/snowpark/_internal/analyzer/datatype_mapper.py

```diff
@@ -91,41 +75,28 @@ def to_sql(value: Any, datatype: DataType, from_values_statement: bool = False)
         # the sql value has to be casted to make sure the varchar length
         # will not be limited.
         return (
-            f"{str_to_sql(value)} :: string"
+            f"{str_to_sql(value)} :: STRING"
             if from_values_statement
             else str_to_sql(value)
         )
-    if isinstance(datatype, ByteType):
-        return f"{value} :: tinyint"
-    if isinstance(datatype, ShortType):
-        return f"{value} :: smallint"
-    if isinstance(datatype, IntegerType):
-        return f"{value} :: int"
-    if isinstance(datatype, LongType):
-        return f"{value} :: bigint"
+
+    if isinstance(datatype, _IntegralType):
+        return f"{value} :: INT"
+
     if isinstance(datatype, BooleanType):
-        return f"{value} :: boolean"
+        return f"{value} :: BOOLEAN"
 
-    if isinstance(value, float) and isinstance(datatype, FloatType):
+    if isinstance(value, float) and isinstance(datatype, _FractionalType):
         if math.isnan(value):
-            cast_value = "'Nan'"
+            cast_value = "'NaN'"
         elif math.isinf(value) and value > 0:
-            cast_value = "'Infinity'"
+            cast_value = "'inf'"
         elif math.isinf(value) and value < 0:
-            cast_value = "'-Infinity'"
+            cast_value = "'-inf'"
```

Comment:
Curious: why do we change this?

Reply:
just make the sql more formal, which are suggested by snowflake doc https://docs.snowflake.com/en/sql-reference/data-types-numeric.html#special-values